Recrutify/Recrutify.Services/Events/StatusChangeEventHandler.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using AutoMapper;
+using Recrutify.DataAccess.Repositories.Abstract;
+using Recrutify.Services.DTOs;
+using Recrutify.Services.Events.Abstract;
+using Recrutify.Services.Services.Abstract;
+
+namespace Recrutify.Services.Events
+{
+    public class StatusChangeEventHandler : IStatusChangeEventHandler
+    {
+        private readonly ICandidateRepository _candidateRepository;
+        private readonly ISendQueueEmailService _sendQueueEmailService;
+        private readonly IProjectService _projectService;
+        private readonly IMapper _mapper;
+        private readonly IUpdateStatusEventArgs _updateStatusEvent;
+
+        public StatusChangeEventHandler(IProjectService projectService, ISendQueueEmailService sendQueueEmailService, ICandidateRepository candidateRepository, IMapper mapper, IUpdateStatusEventArgs updateStatusEvent)
+        {
+            _projectService = projectService;
+            _sendQueueEmailService = sendQueueEmailService;
+            _candidateRepository = candidateRepository;
+            _mapper = mapper;
+            _updateStatusEvent = updateStatusEvent;
+            _updateStatusEvent.UpdateStatusByIdsAsyncComlited += async (e) => await UpdateCandidatesStatusesAsync(e);
+        }
+
+        public async Task UpdateCandidatesStatusesAsync(UpdateStatusEventArgs e)
+        {
+            var candidates = await _candidateRepository.GetByIdsAsync(e.Ids);
+            var project = await _projectService.GetAsync(e.ProjectId);
```

Comment:
you don't need whole project here, make repository method GetNameAsync and return only name (with mongo Project function)

Reply:
i need project.StartDate, what should i do?